File: my-submissions/0908-middle-of-the-linked-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int cnt = 0;
        ListNode* temp = head;
        while(temp != NULL){
            cnt++;
            temp = temp->next;
        }
        if(cnt%2 == 0){
            cnt = cnt/2 +1;
            ListNode* temp1 = head;
        for(int i = 1;i<cnt;i++){
        temp1 = temp1->next;
        }
        return temp1;
        }
        else
        cnt = cnt/2  ;
        ListNode* temp1 = head;
        for(int i = 0;i<cnt;i++){
        temp1 = temp1->next;
        }
        return temp1;
    }
};

File: my-submissions/0908-middle-of-the-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode temp = head;
        int n = 0;
        while(temp != null){
            n++;
            temp = temp.next;
        }
     n = (n/2);
     temp = head;
     while(n >0){
        temp = temp.next;
        n--;
     }
     return temp;

    }
}
